service/migration_manager.cc

```diff
@@ -173,13 +173,16 @@ future<> migration_notifier::unregister_listener(migration_listener* listener)
     return _listeners.remove(listener);
 }
 
-void migration_manager::schedule_schema_pull(const gms::inet_address& endpoint, const gms::endpoint_state& state)
+future<> migration_manager::schedule_schema_pull(const gms::inet_address& endpoint, const gms::endpoint_state& state)
 {
+    auto guard = co_await start_group0_operation();
+    if (guard.with_raft()) {
+        co_return;
+    }
     const auto* value = state.get_application_state_ptr(gms::application_state::SCHEMA);
 
     if (endpoint != utils::fb_utilities::get_broadcast_address() && value) {
-        // FIXME: discarded future
-        (void)maybe_schedule_schema_pull(table_schema_version(utils::UUID{value->value()}), endpoint).handle_exception([endpoint] (auto ep) {
+        co_await maybe_schedule_schema_pull(table_schema_version(utils::UUID{value->value()}), endpoint).handle_exception([endpoint] (auto ep) {
```

Comment:
You are changing semantics here. Previously we pulled in the background. You change it to pull in foreground. Why is it needed?

Reply:
Fixed in https://github.com/scylladb/scylladb/commit/ccd2942567627dce2b55c556b5bc3e48de0deba1